mozfun/event_analysis/extract_event_counts/udf.sql

```diff
@@ -0,0 +1,39 @@
+CREATE OR REPLACE FUNCTION event_analysis.extract_event_counts(events STRING)
+RETURNS ARRAY<STRUCT<index STRING, count INT64>> AS (
+  ARRAY(
+    SELECT AS STRUCT
+      index,
+      COUNT(*) AS count
+    FROM
+      UNNEST(regexp_extract_all(events, r'(.),')) AS index
```

Comment:
Is `index` the right name here? From my reading, it looks like each element of this array is a unicode char representing an event name, rather than a position in the array. Should this be called `event_char`?

Reply:
I might actually prefer that this fall into the general key/value struct paradigm, so would return `ARRAY<STRUCT<key STRING, value INT64>>`. That would make this compatible with `get_key`, etc.

Reply:
In general we've been using the word `index` to describe what the unicode char is that represents an event. I consider it akin to an index in an associative array.

I'm not sure that `udf.get_key` would be useful w.r.t. this udf. The use would be something like: `udf.get_key(event_analysis.extract_event_counts(events), "\U0025")` which should really just be an event count using the view structure.